test/mock_query_test.dart

```diff
@@ -188,6 +188,39 @@ void main() {
     }));
   });
 
+  test('arrayContainsAny', () async {
+    final instance = MockFirestoreInstance();
+    await instance.collection('posts').add({
+      'name': 'Post #1',
+      'tags': ['mostrecent', 'interesting', 'coolstuff'],
+    });
+    await instance.collection('posts').add({
+      'name': 'Post #2',
+      'tags': ['mostrecent'],
+    });
+    await instance.collection('posts').add({
+      'name': 'Post #3',
+      'tags': ['mostrecent'],
+    });
+    await instance.collection('posts').add({
+      'name': 'Post #4',
+      'tags': ['mostrecent', 'interesting'],
+    });
+    instance
+        .collection('posts')
+        .where('tags', arrayContainsAny: ['interesting', 'mostrecent'])
```

Comment:
According to https://firebase.google.com/docs/firestore/query-data/queries#in_and_array-contains-any, array-contains-any should return documents where the value is an array that contains one or more values of the comparison values. So shouldn't it return all 4 documents?

Reply:
Thanks for pointing that out. My bad. I understood Firestore's `array-contains-any` incorrectly. Too bad it doesn't support something like `array-contains-all`, which is what I was looking for.